overlord/configstate/configcore/certs.go

```diff
@@ -32,6 +33,221 @@ import (
+}
+
+func filterCertsInDirectory(dirPath string, blockedCerts []string) error {
+	existingCerts, err := getCertObjects(dirPath)
+	if err != nil {
+		return err
+	}
+
+	// Remove the certs that are marked blocked.
+	for _, cert := range existingCerts {
+		if isBlocked(cert, blockedCerts) {
+			err := os.Remove(cert.Path)
+			if err != nil {
+				return fmt.Errorf("cannot remove blocked certificate: %v", err)
+			}
+		}
+	}
+	return nil
+}
+
+// Populates symbolic links in the output directory, to each certificate
+// provided.
+func installCerts(outputPath string, certs []certificate) error {
+	for _, cert := range certs {
+		// When updating existing certifications we want to make sure we don't
+		// overwrite the certification with itself.
+		certDestinationPath := filepath.Join(outputPath, cert.Name)
+		if certDestinationPath == cert.RealPath {
+			continue
+		}
+
+		// If the path exists, then remove it first to update the certificate
+		if _, err := os.Lstat(certDestinationPath); err == nil {
+			err := os.Remove(certDestinationPath)
+			if err != nil {
+				return fmt.Errorf("cannot remove existing certificate: %v", err)
+			}
+		}
+
+		// Create a link to the real path of the cert in the output path
+		err := os.Symlink(cert.RealPath, certDestinationPath)
+		if err != nil {
+			return fmt.Errorf("cannot copy store certificate: %v", err)
+		}
+	}
+
+	return nil
+}
+
+// Generate the ca-certificates.crt to the output path
+// The ca-certificates.crt is a concatenation of all the certs in the
+// output path.
+func generateCACertificate(outputPath string) error {
+	certificates, err := getCertObjects(outputPath)
+	if err != nil {
+		return err
+	}
+
+	certsPath := filepath.Join(outputPath, "ca-certificates.crt")
+	certsFile, err := os.Create(certsPath)
+	if err != nil {
+		return fmt.Errorf("cannot create ca-certificates.crt: %v", err)
+	}
+
+	for _, cert := range certificates {
+		certBytes, err := ioutil.ReadFile(cert.RealPath)
+		if err != nil {
+			return fmt.Errorf("cannot read certificate %q: %v", cert.Name, err)
+		}
+
+		_, err = certsFile.Write(certBytes)
+		if err != nil {
+			return fmt.Errorf("cannot write certificate %q: %v", cert.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func getCertObjectsFromPaths(outputPath string, inputPaths []string) ([]certificate, error) {
+	certObjects := []certificate{}
+	for _, certDirectoryPath := range inputPaths {
+		// If the input directory equals output, then ignore it
+		certFullPath, err := filepath.Abs(certDirectoryPath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot get absolute path for path: %v", certDirectoryPath)
+		}
+		if certFullPath == outputPath {
+			continue
+		}
+
+		certs, err := getCertObjects(certDirectoryPath)
+		if err != nil {
+			return nil, fmt.Errorf("cannot read certs: %v", err)
+		}
+		certObjects = append(certObjects, certs...)
+	}
+	return certObjects, nil
+}
+
+// Allows the caller to combine multiple certification directories into one single directory.
```

Comment:
Nitpick, per go conventions the doc string should start with function name.

Reply:
Thank you in any case, I'm still learning go and how we do things so all this 'nitpicking' is really appreciated!